test/benchmark/reasoner/iam/basic/BasicTest.java

```diff
@@ -160,6 +160,24 @@ public void testDoubleJoin() {
         benchmark.assertCounters(25, NOBJECTS * NOBJECTS, NOBJECTS, NOBJECTS + 3);
     }
 
+    @Test
+    public void testTripleJoinWithProjection() {
+        String query = String.format(
+                "match\n" +
+                        "$a isa object, has id \"%s\";\n" +
+                        "(start: $a, end: $b) isa object-pair;\n" +
+                        "(start: $b, end: $c) isa object-pair;\n" +
+                        "(start: $c, end: $d) isa object-pair;\n" +
+                        "get $a, $d;\n",
+                queryParams.basicTestObject);
+        Benchmark benchmark = new Benchmark("double-join", query, NOBJECTS);
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(1500);
+        benchmark.assertCounters(25, NOBJECTS * NOBJECTS, NOBJECTS, NOBJECTS + NOBJECTS + 3);
+    }
```

Comment:
Minimal example to show why the re-usability is useful.

TODO: See if I can add one to create a ring so the stream-graph restructuring is tested.